File: my-dashboard/test_data_gov.js
async function testDataGov() {
  try {
    const res = await fetch("https://data.gov.tw/dataset/4523");
    const html = await res.text();
    // Search for href containing '.csv' or '.json' or 'Download.ashx'
    const matches = html.match(/href="([^"]*)"/g);
    const links = matches.map(m => m.replace(/href="|"/g, ''));
    const zipLinks = links.filter(l => l.includes('.zip') || l.includes('zip'));
    console.log("ZIP links:", zipLinks);
    
    // Check if there is an API button or link
    const apiLinks = links.filter(l => l.includes('api') && l.includes('ndc'));
    console.log("API links:", apiLinks);
    
  } catch(e) { console.error(e); }
}
testDataGov();

File: my-dashboard/test_data_gov.py
import requests
from bs4 import BeautifulSoup
import json

def get_data():
    url = "https://data.gov.tw/dataset/4523"
    headers = {"User-Agent": "Mozilla/5.0"}
    res = requests.get(url, headers=headers)
    soup = BeautifulSoup(res.text, "html.parser")
    # find all a tags where href contains Download.ashx
    links = [a['href'] for a in soup.find_all('a', href=True) if 'Download.ashx' in a['href'] or 'ndc.gov.tw' in a['href']]
    print("Links found:", links)

if __name__ == "__main__":
    get_data()
